pyanaconda/anaconda.py

```diff
@@ -165,9 +165,19 @@ def storage(self):
         if not self._storage:
             import blivet
             self._storage = blivet.Blivet(ksdata=self.ksdata)
-
-            if self.instClass.defaultFS:
-                self._storage.setDefaultFSType(self.instClass.defaultFS)
+            fstype = None
+
+            # Get the default fstype from a kickstart file.
+            if self.ksdata.autopart.autopart and self.ksdata.autopart.fstype:
+                fstype = self.ksdata.autopart.fstype
+            # Or from an install class.
+            elif self.instClass.defaultFS:
+                fstype = self.instClass.defaultFS
+
+            # Set the default fstype.
+            if fstype:
+                self._storage.setDefaultFSType(fstype)
+                self._storage.setDefaultBootFSType(fstype)
```

Comment:
What about creating new method `self._set_default_fstype()` and moving this to the new method?